src/nevergreen/gateway.clj

```diff
@@ -61,25 +66,28 @@
 (defn- http [url data method]
   (let [redacted-url (redact-url url)]
     (log/info (str "Calling [" redacted-url "]..."))
-    (try
-      (let [res (method url (merge {:insecure?             true
-                                    :socket-timeout        ten-seconds
-                                    :conn-timeout          ten-seconds
-                                    :throw-entire-message? true
-                                    :cookie-policy         :standard}
-                                   data))]
-        (log/info (str "[" redacted-url "] returned a status of [" (:status res) "]"))
-        (:body res))
-      (catch SocketTimeoutException e
-        (handle-timeout url redacted-url e))
-      (catch UnknownHostException e
-        (handle-unknown-host url redacted-url e))
-      (catch URISyntaxException e
-        (handle-invalid-url url redacted-url e))
-      (catch ConnectException e
-        (handle-connection-refused url redacted-url e))
-      (catch ExceptionInfo e
-        (handle-exception-info url redacted-url e)))))
+    (let [future (method url (merge {:insecure?             true
+                                     :socket-timeout        ten-seconds
+                                     :conn-timeout          ten-seconds
+                                     :throw-entire-message? true
+                                     :cookie-policy         :standard
+                                     :async                 true}
+                                    data))]
+      (try
+        (let [res (.get future 1 TimeUnit/MINUTES)]
```

Comment:
Fortunately, we did not need any library here. executing requests asynchronously returns a **Future** and can pass a certain timeout value for the result to be computed, It it crosses that threshold a `TimeoutException` will be raised which can be handled and **cancel** the request to prevent CI server overload.